compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3961,6 +3961,26 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         if let Node::Expr(expr) = tcx.hir_node(arg_hir_id)
             && let Some(typeck_results) = &self.typeck_results
         {
+            if let hir::Expr { kind: hir::ExprKind::MethodCall(_, rcvr, _, _), .. } = expr
+                && let Some(ty) = typeck_results.node_type_opt(rcvr.hir_id)
+                && let Some(failed_pred) = failed_pred.to_opt_poly_trait_pred()
+                && let pred = failed_pred.skip_binder().with_self_ty(tcx, ty)
```

Comment:
use `map_bound` here. otherwise, it's gonna ICE because you're skipping binders.